### PR TITLE
Renaming Earn features

### LINKED ISSUE #42860: Block: Update Recurring Payments Block to offer one time payments
Part of https://github.com/Automattic/wp-calypso/issues/42825

Let's update the recurring payments **block** options to allow one time payments. See also pbMlHh-e1-p2

There is an existing issue https://github.com/Automattic/wp-calypso/issues/42858 to cover WordPress.com payments plan settings.

### PR DESCRIPTION
**Pull request by eoigal**
Fixes https://github.com/Automattic/wp-calypso/issues/42860
Part of https://github.com/Automattic/wp-calypso/issues/42861
Part of https://github.com/Automattic/wp-calypso/issues/42862

#### Changes proposed in this Pull Request:

* Adds a one-time payment feature to Recurring Payments and rename the entire Recurring Payments feature to “Payments”. The bet is that this will simplify the user experience and increase usage, as customers will now have only one place to go when they want to accept payments.
* “Simple Payments” as a feature will still be available to customers already utilizing it, but it's renamed to "Pay with PayPal"

#### Is this a new feature or does it add/remove features to an existing part of Jetpack?

pbMlHh-hn-p2

#### Does this pull request change what data or activity we track or use?
Not applicable.

#### Testing instructions:

* The Simple Payments block should now be called Pay with PayPal - verify this by creating a post with block editor and adding a Pay with PayPal block.

icon
shortcut
esc-block
block-description

* Make sure the block still appear in the block inserter when searching for "Simple Payments".

* Confirm that the Simple Payments widget is also named Pay with PayPal.
widget-form
widget-title

* Verify the Recurring Payments block has been renamed to Payments.

Zrzut ekranu 2020-06-8 o 16 46 44

Zrzut ekranu 2020-06-8 o 16 46 56

* Create a new payments plan, with the "One time" option.
* Publish the post and visit it while logged out.
* Go through checkout.

Zrzut ekranu 2020-06-8 o 16 48 11

#### Proposed changelog entry for your changes:
Rename Earn features (Simple Payments > Pay with PayPal, Recurring Payments > Payments).

### CODE REVIEW

**Review by mmtr — COMMENTED** [review]
Thanks for preparing the support documentation, @eoigal! This works great.

But I think we missed one occurrence in the new Jetpack Setup Wizard that was recently merged in https://github.com/Automattic/jetpack/pull/15637.

https://github.com/Automattic/jetpack/blob/72a90143f45055aa5be1ea6cedd1d91e5ae6739e/_inc/client/setup-wizard/feature-toggle/map-feature-to-props.js#L884

**Comment by mmtr** [comment]
> This change will require changes to support articles. I've created the following articles that duplicate the existing Simple Payments support articles;


Just found out one more article we'd need to update: https://jetpack.com/support/jetpack-blocks/

Screen Shot 2020-06-11 at 17 38 27

**Comment by gwwar** [comment]
There's another usage in https://github.com/Automattic/jetpack/blob/72a90143f45055aa5be1ea6cedd1d91e5ae6739e/_inc/client/setup-wizard/feature-toggle/map-feature-to-props.js#L884-L885

Though I don't think I've seen this wizard in action before

**Comment by gwwar** [comment]
From testing instructions in https://github.com/Automattic/jetpack/pull/15637

we should be able to see the wizard with:
* Add the following to the end of wp-config.php: add_filter( 'jetpack_show_setup_wizard', '__return_true' ); (Not sure if we still need this step)
* Visit /wp-admin/admin.php?page=jetpack#/setup/features.

**Review by mmtr — DISMISSED** [review]
LGTM

**Review by jeherve — COMMENTED** [review]
> Pay with PayPal button

We moved away from the word "button" in the block name for Recurring Payments and Simple Payments in #15061, following feedback from @mtias in p58i-8JL-p2

Should we stick with "Pay with Paypal" only for the block name?

**Comment by mtias** [comment]
It does look like a really long name. The extra "button" text seems superfluos as @jeherve points out. Also curious why just "PayPal" was discarded as an option?

**Comment by artpi** [comment]
> It does look like a really long name. The extra "button" text seems superfluos as @jeherve points out. Also curious why just "PayPal" was discarded as an option?

There are legal considerations: p2y3YZ-43P-p2

However I am curious @lessbloat @gwwar  - do we really need the "Button" part?

**Comment by eoigal** [comment]
Removed use of `button` where I think its superflous in D44948-code

**Comment by gwwar** [comment]
> However I am curious @lessbloat @gwwar - do we really need the "Button" part?

Not sure, I'm checking in with folks.

**Comment by kwight** [comment]
@jeherve any suggestion for how best to apply these changes conditionally _after_ merge to master, but _before_ the plugin release? Meaning, we would merge the dotcom patch soon also, but not "flip the switch" before the release on July 7th?

I'd normally think a filter, but it's super awkward when dealing on a string-by-string basis. Would we be OK with a constant check that we then remove once the plugin is out?

```php
title: is_defined( 'WPCOM_PAY_WITH_PAYPAL_ACTIVE' ) ? __( 'Pay with Paypal' ) : __( 'Simple Payments Block' ),
```

**Comment by jeherve** [comment]
> any suggestion for how best to apply these changes conditionally after merge to master, but before the plugin release? Meaning, we would merge the dotcom patch soon also, but not "flip the switch" before the release on July 7th?

Until now, we've just held off committing matching WordPress.com diffs until an hour before the release whenever a diff would introduce changes to block markup or other important changes like that. I keep a list of such diffs and usually commit them myself or coordinate with the PR author so they commit themselves on release day.

**Review by jeherve — CHANGES_REQUESTED** [review]
The tests appear to be failing right now.

**Comment by gwwar** [comment]
We should be able to use "Pay with PayPal" without the button. I'll update related issues.

**Comment by gwwar** [comment]
Let's also make sure that search teams for "Simple Payments" will still return the block, in addition to the new block name.

File: modules/widgets/simple-payments/admin-warning.php
Comment (gwwar) [inline modules/widgets/simple-payments/admin-warning.php:5]: We may want to update the doc permalink and this url to `https://wordpress.com/support/pay-with-paypal`

File: modules/widgets/simple-payments/form.php
Comment (gwwar) [inline modules/widgets/simple-payments/form.php:3]: Just "Pay with Paypal"

File: extensions/blocks/simple-payments/index.js
Comment (gwwar) [inline extensions/blocks/simple-payments/index.js:69]: Nice, thanks for adding older search terms @mmtr !

File: extensions/blocks/recurring-payments/edit.jsx
Comment (gwwar) [inline extensions/blocks/recurring-payments/edit.jsx:81]: I think we want to keep the default to monthly, based on https://github.com/Automattic/wp-calypso/pull/42848#issuecomment-639594153

**Review by gwwar — DISMISSED** [review]
Thanks @mmtr for continuing with this one! Overall this one tests well for me! So is the plan to merge this one, but hold off on the wpcom side until July 7th?

Some things to confirm before merge:
- I had some trouble getting the PayPal widget to display. (Was it linked to a higher plan than premium?) 
- I think we may need to rebase with master

### Pay with PayPal
Screen Shot 2020-06-17 at 4 21 24 PM
Screen Shot 2020-06-17 at 4 21 58 PM
Screen Shot 2020-06-17 at 4 22 07 PM

### Payments + One time option
Screen Shot 2020-06-17 at 4 33 53 PM
Screen Shot 2020-06-17 at 4 33 58 PM
Screen Shot 2020-06-17 at 4 36 12 PM
Screen Shot 2020-06-17 at 4 37 33 PM
Screen Shot 2020-06-17 at 4 37 46 PM

**Comment by gwwar** [comment]
This one also fixes https://github.com/Automattic/wp-calypso/issues/42860

**Comment by mmtr** [comment]
> So is the plan to merge this one, but hold off on the wpcom side until July 7th?

Correct. I'm happy to make sure the diffs are updated so they can be applied cleanly on July 7.

> I had some trouble getting the PayPal widget to display. 

Yeah, I'm not able to find it as well (at least testing on a JN site). According to docs it should be in Appearance > Customize > Widgets on sites under a Premium plan or higher. @jeherve do you have any clue about what could be happening?

**Review by jeherve — DISMISSED** [review]
This tests well for me, it should be good to merge.

> Yeah, I'm not able to find it as well (at least testing on a JN site). According to docs it should be in Appearance > Customize > Widgets on sites under a Premium plan or higher. @jeherve do you have any clue about what could be happening?

You would need to go to Jetpack > Settings and enable the Extra Sidebar Widgets there in order to see all the widgets Jetpack has to offer.

I don't think that's very user-friendly, especially since the toggle is only available if you search for it, but I think that's part of a larger discussion around the visibility of our features.

**Comment by gwwar** [comment]
Oh thanks @jeherve! Name changes also look good in the widget:

Screen Shot 2020-06-18 at 10 56 47 AM
Screen Shot 2020-06-18 at 10 57 16 AM
Screen Shot 2020-06-18 at 10 57 26 AM

**Comment by mmtr** [comment]
I think we may want to hold off merging this PR until we have the Recurring Payments support docs renamed too, so we can update the links ([example](https://github.com/Automattic/jetpack/blob/f7ec966a2206e8a723a666c0c8cbd2a9cb557853/extensions/blocks/recurring-payments/edit.jsx#L381)) as well in this PR and therefore keep the uncommitted WP.com changes in one single diff.

**Comment by mmtr** [comment]
> I think we may want to hold off merging this PR until we have the Recurring Payments support docs renamed too, so we can update the links ([example](https://github.com/Automattic/jetpack/blob/f7ec966a2206e8a723a666c0c8cbd2a9cb557853/extensions/blocks/recurring-payments/edit.jsx#L381)) as well in this PR and therefore keep the uncommitted WP.com changes in one single diff.

That example turned out to be the only link to the Recurring Payments support docs we have in the block. I created a duplicated version of it removing any Recurring usage and updated the link in 87b9503, so this should be good to merge now.

**Review by jeherve — APPROVED** [review]

**Comment by jeherve** [comment]
r210097-wpcom

**Pull request merged**
